core/benches/sigverify_stage.rs

```diff
@@ -170,18 +171,21 @@ fn bench_sigverify_stage(bencher: &mut Bencher) {
         }
         let mut received = 0;
         trace!("sent: {}", sent_len);
+        let mut messages = vec![];
         loop {
-            if let Ok((mut verifieds, _)) = verified_r.recv_timeout(Duration::from_millis(10)) {
-                while let Some(v) = verifieds.pop() {
+            if let Ok(message) = verified_r.recv_timeout(Duration::from_millis(10)) {
+                let (verifieds, _) = &*message;
+                for v in verifieds.iter().rev() {
```

Comment:
it was a leftover to keep semantics with old code as much as possible to avoid review churn like this... `.rev()` was matched to the `.pop()`.

yeah, i failed at that. lol

anyway, as you spotted, `received += ...` in the reverse order doesn't make sense at all.

Originally, I was too lazy to read the motto of these touched benches/tests. seems it was overdue. After serious code read, I found numerous problems and settled on this final state....: 7b930d9ed7bcfdc39a1167d7a468cb22eff0301d 0b72244a9b7b501e337b59fc5091123c30fd1eef 88e261cb3c0dc21ad8e48e4d41b36d63dc20a37c

Anyway, that `.rev()` is removed altogether.

Reply:
just refs (tldr: these code were bitrot to some extent. i barely made them work):

https://github.com/solana-labs/solana/pull/4320
https://github.com/solana-labs/solana/pull/23066
https://github.com/solana-labs/solana/pull/25359